### PR TITLE
Use API to synchronise with JWPlatform

### LINKED ISSUE #12: Only import audio if that is the source format
.. ie it isn't extracted from source video as a seperate clip

### LINKED ISSUE #13: Pruning
- prune any JW media items that don't match the latest RSS feed (eg. old audio items)
- prune any custom tags that (eg sms_visibility) are no longer required.

depends: #12

### PR DESCRIPTION
**Pull request by rjw57**
> **IMPORTANT:** This PR requires the 1.2.2 release of jwplayer's client. Track progress of this at https://github.com/jwplayer/jwplatform-py/pull/24.

Instead of using MRSS feeds to attempt JWPlatform synchronisation, move to using the API directly.

This PR generalises applyupdatejob and genupdatejob to perform full synchronisation of videos:

1. applyupdatejob has been taught how to create and delete resources in addition to updating them.
2. genupdatejob looks through the current SMS database and generates create jobs for missing videos.
3. Add a new command, tidy, which determines which videos need to be deleted. See the documentation for details.

With the addition of create jobs, we no longer need to generate feeds.

The tidy command has been split from genupdatejob because tidy is the only command which generates *delete* jobs and I'd like to keep a separation between "safe" (update only) and "risky" (can delete) commands.

**NOTE:** Thumbnails are not added in this PR. It's a little subtle to work out if a particular video has a custom thumbnail or not with the JWPlatform API. At first look it appears that there is no way to tell. We may be able to use some sort of custom prop hack to track this ourselves but I'm going to punt that to a different story.

Closes #12. Closes #13.

### CODE REVIEW

File: sms2jwplayer/__init__.py [hunk 1]
```diff
@@ -43,6 +44,7 @@
     fetch               Fetch details on all videos in jwplayer.
     genupdatejob        Generate list of missing metadata for each video key.
```
Comment (msb) [inline sms2jwplayer/__init__.py:45]: "applyupdatejob" is missing

**Review by msb — COMMENTED** [review]
All good - other than you may want to consider adding the ability to prune custom properties that no longer exist.

File: sms2jwplayer/__init__.py (hunk 1 above)
Comment (rjw57) [reply sms2jwplayer/__init__.py:45]: Good catch. Will fix.

**Comment by rjw57** [comment]
> All good - other than you may want to consider adding the ability to prune custom properties that no longer exist.

I think properties which end up blanked in the SMS DB should also be blanked on the corresponding JWPlayer video. (If not, then that's a bug). However, I intentionally don't touch any properties that I don't know about to allow us to have more custom props going forward.

**Review by msb — APPROVED** [review]

**Pull request merged**